### DISCUSSION TITLE
Ability to control oxygen flow in rooms? (via Status Monitor for example)

### DISCUSSION THREAD

**Discussion by Stallfighter** [FakeFishGames/Barotrauma#5004] [Ideas and suggestions]
It may be useful in fight with human enemies (and/or mutinies...) and big fires without cutting O2Gen completely. Basically, like it was in good ol' FTL minus remote door control (well you can wire that up too if you really want to...)
Yes, you can use pumps to do something similar but it messes up with our sub buoyancy.

I think it may be added to Status Monitor functionality specifically so that station sits there not only to just watch O2/hull/water status (and on-board alien liveform detector with some modifications).

**Comment by mLuby** [comment]
You _can_ achieve this in Vanilla by relaying power to a hidden O2 generator linked to a single vent. ([Example](https://steamcommunity.com/sharedfiles/filedetails/?id=2388296229)) That way, disabling the relay disables the generator, which cuts off oxygen from that vent. Connecting multiple relayed vents to the "main" O2 generator gives the illusion that all the O2 is being supplied by that one machine. It'd be cleaner if vents had toggle/set_state pins (see this ["Links as wires" proposal](https://github.com/Regalis11/Barotrauma/discussions/6961)), but IMO this workaround isn't terrible.

![o2 vent](https://user-images.githubusercontent.com/2483420/107398142-57465880-6acd-11eb-98e8-c678f37096f9.gif)

Custom lights, labels, and buttons in the Status Monitor would be nice, but should probably be a separate discussion.

**Reply by Stallfighter** [reply]
Well we can get AT LEAST logical wiring to vents to on/off it via buttons which is a little bit bulky but still makes the job done

**Comment by Dave247** [comment]
> It'd be cleaner if vents had toggle/set_state pins,

This is what the suggestion point really boils down to. If you wanted to be really fancy you could still have more then one generator though if you wanted to split it between the ship and refilling oxygen tanks, while removing the need for any complex relay wiring systems.
